Core/Matcher/ContentMatcher.php

```diff
@@ -44,6 +44,21 @@ public function matchContent(array $conditions)
     {
         $this->validateConditions($conditions);
 
+        if (count($conditions) === 1) {
+            $key = array_keys($conditions)[0];
+            switch ($key) {
+                case self::MATCH_CONTENT_ID:
+                    $location = $this->repository->getContentService()->loadContent($conditions[$key]);
```

Comment:
1. I think that we support both single ids and arrays of ids - both here and for location. Maybe it would be better to load stuff using a Matcher?
2. while at it, we could apply the same logic of direct loading for load-by-remote-id?

Reply:
ok, nvm point 2. Temporary blindness :-D